pkg/machine/machine.go

```diff
@@ -0,0 +1,61 @@
+package machine
+
+import "errors"
+
+type State interface {
+	~string
+}
+
+// Configuration maps where a from state can transition to
+type Configuration[S State] struct {
```

Comment:
Can this be called like Allowable or Possible Transitions?

Reply:
changed this to allowable and some of the fields to match what they are better